lib/Alchemy/Phrasea/Model/Repositories/BasketElementRepository.php

```diff
@@ -79,6 +79,29 @@ public function findElementsByRecord(\record_adapter $record)
         return $query->getResult();
     }
 
+    public function findElementsDatasByRecord(\record_adapter $record)
+    {
+        $dql = <<<'DQL'
+SELECT be
+FROM Phraseanet:BasketElement AS be
+INNER JOIN be.basket b
+INNER JOIN b.validation AS vs
+INNER JOIN vs.participants AS vp
+INNER JOIN vp.datas AS vd 
+WHERE be.record_id = :record_id
+AND be.sbas_id = :sbas_id
+ORDER BY b.created DESC
+DQL;
+
+        $query = $this->_em->createQuery($dql);
+        $query->setParameters([
+            'sbas_id' => $record->getDataboxId(),
+            'record_id' => $record->getRecordId(),
+        ]);
+
+        return $query->getResult();
+    }
+
```

Comment:
**TO BE DISCUSSED WITH NM :** 
maybe we could filter directly in dql with rules : 
- is creator of the validation
or
- is participant, with "canseeothers"
so we could avoid ACL::CANMODIFRECORD in twig

todo : **check by sql before coding**